### PR TITLE
Upgrade ElasticSearch to 0.4

### PR DESCRIPTION
**Pull request by mwl**
Fixes mesos/elasticsearch#237

### CODE REVIEW

File: repo/packages/E/elasticsearch/1/marathon.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    "--executorName={{elasticsearch.executor.name}}",
+    "--executorTimeout={{elasticsearch.executor.timeout}}",
+  ],
+  "env": {
+    "JAVA_OPTS": "{{elasticsearch.scheduler.java-heap}}"
+  },
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "mesos/elasticsearch-scheduler",
```
Comment (philwinder) [inline repo/packages/E/elasticsearch/1/marathon.json]: Make this a parameter, and set the default to "mesos/elasticsearch-scheduler:0.3.0"

File: repo/packages/E/elasticsearch/1/config.json [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+              "description":"URI to ES yml settings file. If file is copied to all slaves, the file must be in /tmp/config. E.g. 'file:/tmp/config/elasticsearch.yml', 'http://webserver.com/elasticsearch.yml'",
+              "type":"string"
+            },
+            "health-delay":{
+              "default":"3000",
+              "description":"The delay between executor healthcheck requests (ms).",
+              "type":"string"
+            },
+            "image":{
+              "default":"mesos/elasticsearch-executor",
```
Comment (philwinder) [inline repo/packages/E/elasticsearch/1/config.json]: Make this a tagged version :0.3.0, so when we update the images, this doesn't break.

File: repo/packages/E/elasticsearch/1/config.json [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+              "description":"The name given to the executor task.",
+              "type":"string"
+            },
+            "timeout":{
+              "default":"60000",
+              "description":"The maximum executor healthcheck timeout (ms). Must be greater than elasticsearch.executor.healthdelay. Will start new executor after this length of time.",
+              "type":"string"
+            }
+          },
+          "required":[
```
Comment (philwinder) [inline repo/packages/E/elasticsearch/1/config.json]: Ah, another problem. Aparrently the required parameter, doesn't actually mean this parameter is required by the user. It means it is required by the config parsing frameworks. 
I didn't really understand this, but basically it means that all parameters need to be in their respective required array. E.g. timeout should be here, etc.

**Comment by mwl** [comment]
Waiting for https://hub.docker.com/r/mesos/elasticsearch-scheduler/ version 0.3.0 to be released

**Comment by frankscholten** [comment]
@mwl 0.4.0 released yesterday

**Comment by mwl** [comment]
Input from team
>  Idea from James regarding tasks task list. Why not remove all tasks lists from scheduler completely. When THe GUI calls the rest endpoint in the controller, it just populates a new list by constructing a ClusterState object and returning the task list. I.e. on every call it just reads zookeeper. No state in scheduler.
> I also found a bug. You set the Tasks info at start, and when an executor is created. But it is never updated. It does not get updated when we get a status update.
This won't be a problem if you rewrite using James' idea above.

**Comment by mwl** [comment]
@kensipe We're done upgrading the framework to 0.4.1. So it's ready for you to look at.

**Comment by kensipe** [comment]
I will take a look by tomorrow... however afterward, this work may get moved to the multiverse which is our incubator for services.  This seems like it belongs there.     I'll let you know.

**Comment by kensipe** [comment]
@mwl `dcos package install elasticsearch` results in an unhappy framework.

```
I0923 00:47:43.486145  2027 exec.cpp:207] Executor registered on slave 20150922-233807-1409744906-5050-1301-S0
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:53)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.beust.jcommander.ParameterException: Parameter --zookeeperFrameworkUrl cannot be empty
	at org.apache.mesos.elasticsearch.common.cli.validators.CLIValidators$NotEmptyString.validate(CLIValidators.java:64)
	at com.beust.jcommander.ParameterDescription.validateParameter(ParameterDescription.java:300)
	at com.beust.jcommander.ParameterDescription.validateParameter(ParameterDescription.java:268)
	at com.beust.jcommander.ParameterDescription.addValue(ParameterDescription.java:237)
	at com.beust.jcommander.ParameterDescription.addValue(ParameterDescription.java:209)
	at com.beust.jcommander.JCommander.processFixedArity(JCommander.java:875)
	at com.beust.jcommander.JCommander.processFixedArity(JCommander.java:857)
	at com.beust.jcommander.JCommander.parseValues(JCommander.java:725)
	at com.beust.jcommander.JCommander.parse(JCommander.java:282)
	at com.beust.jcommander.JCommander.parse(JCommander.java:265)
	at org.apache.mesos.elasticsearch.scheduler.Configuration.<init>(Configuration.java:39)
	at org.apache.mesos.elasticsearch.scheduler.Main.run(Main.java:26)
	at org.apache.mesos.elasticsearch.scheduler.Main.main(Main.java:20)
	... 6 more
```

**Comment by kensipe** [comment]
@mwl any updates?

**Comment by mwl** [comment]
@kensipe Found the culprit. Please try again.

**Comment by kensipe** [comment]
awesome thx!

**Comment by kensipe** [comment]
@mwl this is not working.. it appears to be a zk issue (it looks like @jgarcia-mesosphere PR fixes it)  after adding that fix to this... it fails after zk and master initialization.  with:  

Starting task elasticsearch.e9e2096c-6946-11e5-a32c-46ab80f7ae7e
Killing docker task
Shutting down

I was looking to merge this prior to moving to moving it to multiverse... at this point... I'm going move the existing elasticsearch.  please send a PR there.

**Pull request closed without merging**